templates/zerver/help/connect-through-a-proxy.md

```diff
@@ -0,0 +1,77 @@
+# Connect through a proxy
+
+Some corporate and university networks may require you to connect to Zulip
+via a proxy.
+
+## Web
+
+Zulip uses your browser's default proxy settings. To set a custom proxy just
+for Zulip, check your browser's instructions for setting a custom proxy for
+a single website.
+
+## Desktop
+
+1. Click on the **gear** (<i class="icon-vector-cog"></i>) icon in the lower left corner.
+
+2. Select the **Network** tab.
+
+### System proxy settings
+
+3. Click **Use system proxy settings**.
+
+4. Restart the Zulip desktop app.
+
+### Custom proxy settings
+
+3. Click **Manual proxy configuration**.
+
+4. Either enter a URL for **PAC script**, or fill out **Proxy rules** and
+  **Proxy bypass rules**.
+
+5. Click **Save changes**.
+
+In most corporate environments, your network administrator will provide a
+URL for the **PAC script**.
+
+The second most common configuration is that your network administrator has
+set up a proxy server for accessing the public internet, but URLs on the
+local network must be accessed directly. In that case set **Proxy rules** to
+the URL of the proxy server (it may look something like
+`http://proxy.example.edu:port`), and **Proxy bypass rules** to cover local URLs
+(it may look something like `*.example.edu,10.0.0.0/8`).
+
+If either of those apply, you can skip the rest of this guide. If not, we
+document the full syntax for **Proxy rules** and **Proxy bypass rules**
+below.
+
+#### Proxy rules
+
+A semicolon-separated list of `protocolRule`s.
+
+```
+protocolRule -> [<protocol>"="]<URLList>
+protocol -> "http" | "https" | "ftp" | "socks"
+URLList -> comma-separated list of URLs
+```
+
+Some examples:
+
+* `http=http://foopy:80;socks=http://foopy2:1080` - Use proxy `http://foopy:80`
```

Comment:
`socks=http://foopy2:1080` seems wrong (though this is practically possible). I think it'd be better suited to put something like `socks=foopy2:1080` or `socks=socks5://foopy2:1080`.

Reply:
My understanding of our previous conversation was that `socks=http://foopy2:1080` is equivalent to `socks=foopy2:1080`? Or did I misunderstand?

Reply:
The electron proxy [docs](https://github.com/electron/electron/blob/master/docs/api/session.md#sessetproxyconfig-callback) is kind of confusing as to what is what :sweat_smile:.
My understanding from the docs is that `socks=foopy2:1080` is equivalent to `socks=socks4://foopy2:1080`. Previously we have had users who have successfully used socks proxy using `socks=socks5://foopy2:1080` . I think a `socks` proxy url having a prepending `socks5://` would be the general case.

Reply:
I see, the docs are indeed pretty confusing.

Reply:
I think both convey the same idea, so if `socks=http://...` looks wrong to people I think it's fine to use the ftp example. I'm fine with either, though

Reply:
How about `socks=socks5://foopy:1080` using Socks v5 proxy?